src/components/cards/MarketsCard.tsx

```diff
@@ -139,11 +139,20 @@ export const MarketsCard = (props: Props) => {
   const coingeckoFiat = useSelector(state => getCoingeckoFiat(state))
 
   const [coinRankingDatas, setCoinRankingDatas] = React.useState<CoinRankingData[]>([])
+  // Detect change to fiat to allow refresh
+  const [lastFetchedFiat, setLastFetchedFiat] = React.useState<string>(coingeckoFiat)
```

Comment:
A clearer approach to this commit to fix the issue.

First, make the async function for the periodic task a ref so it can depend on the most current state in the component (`coingeckoFiat` and `numRows`). This way the periodic task is always working with the most current state when executing it's routine.

Second, make the effect only depend on `coingeckoFiat` so that it only resubscribes when this changes.

Third, always reset the `coinRankingDatas` state before setting up the periodic task subscription for the component. 

With this approach, there is no need for the extra state `lastFetchedFiat` and we get less churn on the periodic task subscription.